### PR TITLE
feat: Add next-auth middleware to protect pages

### PR DESCRIPTION
**Pull request by flacial**
## Motivation

At present, we do not have a streamlined approach to prevent unauthorized access to pages that require the user to be logged in. The current implementation involves manual checks on the session to determine if the user needs to be redirected to the login page or not.

## Solution

Introduce `next-auth` middleware to handle the protection of pages that require the user to be logged in. The middleware will maintain a list of protected paths and in case the user is not logged in, they will be immediately redirected to the `/login` page. This not only simplifies the codebase by removing the redundant code for page protection but also saves bandwidth for the user.

## Changes

- Implement `next-auth` middleware to protect the `/settings/account` page
- Create a constant for the path `/settings/account`
- Set the redirection to `/login` for unauthenticated users
- [Remove the `secret` option in favor of using `NEXTAUTH_SECRET` in Vercel's project environment variables](https://next-auth.js.org/configuration/options#nextauth_secret)
- Add a test file to cover the `middleware.ts` file

## Testing

1. Log out of the application
2. Access the `/settings/account` page and verify if it redirects to `/login`
3. Log in to the application by filling in the login form
4. Verify if it redirects to `/settings/account` after successful login

## Additional Info

In future pull requests, more pages such as `/review/[lessonSlug]` will be added to the list of protected pages.

## Related issues

- #1629

### CODE REVIEW

**Comment by SlyBouhafs** [comment]
I noticed there are some hardcoded paths in the new changes, maybe consider using the constants instead?

File: __tests__/pages/api/auth/[...nextauth].test.js [hunk 1]
```diff
@@ -26,7 +26,9 @@ describe('next-auth middleware', () => {
         jwt,
         session
       },
-      secret: process.env.SESSION_SECRET
+      pages: {
+        signIn: '/login'
```
Comment (SlyBouhafs) [inline __tests__/pages/api/auth/[...nextauth].test.js:30]: `LOGIN_PATH` here?

File: middleware.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export { default } from 'next-auth/middleware'
+
+export const config = { matcher: ['/settings/account'] }
```
Comment (SlyBouhafs) [inline middleware.ts:3]: Use the new `SETTINGS_ACCOUNT_PATH` constant here?

File: pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -15,5 +15,7 @@ export default (
       jwt,
       session
     },
-    secret: process.env.SESSION_SECRET
+    pages: {
+      signIn: '/login'
```
Comment (SlyBouhafs) [inline pages/api/auth/[...nextauth].ts:19]: Same thing here, `LOGIN_PATH`?

File: pages/login.tsx [hunk 4]
```diff
@@ -153,8 +153,11 @@ const LoginPage: React.FC<GetAppProps> & WithLayout = ({
 
     if (ok && !error) {
       window.localStorage.setItem('loggedIn', 'true')
-      const { next } = router.query
-      router.push(next ? (next as string) : '/curriculum')
+      const { next, callbackUrl } = router.query
+      const path = next || callbackUrl
+      const redirectTo = path || '/curriculum'
```
Comment (SlyBouhafs) [inline pages/login.tsx:158]: We can use `CURRICULUM` here?

**Comment by flacial** [comment]
@SlyBouhafs Thanks for reviewing the PR :)

I fixed most of the issues.

**Review by SlyBouhafs — APPROVED** [review]

**Pull request merged**